common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -259,6 +269,14 @@ protected enum ImplementationProvider {
         UTF8_BINARY, ICU
       }
 
+      /**
+       * Bits 19-18 having value 00 for no space trimming, 01 for left space trimming
+       * 10 for right space trimming and 11 for both sides space trimming.
```

Comment:
if we end up going with this, I would just add a note in this comment that 1 bit still remains reserved at this time
that is, if we don't go with: https://github.com/apache/spark/pull/48222/files#r1774931595

Reply:
Sure, will update.